server/src/test/java/io/unitycatalog/server/utils/PopulateTestDatabase.java

```diff
@@ -278,59 +279,53 @@ public static void main(String[] args) throws JsonProcessingException {
-            .build();
-    addTypeTextAndJsonText(country);
+
     String partitionedTableName = "user_countries";
     String partitionedStorageRoot = "etc/data/external/";
     String partitionedTablePath = partitionedStorageRoot + catalogName + "/" + schemaName + "/tables/" + partitionedTableName;
-    UUID partitionedTableId = UUID.randomUUID();
-
-    TableInfoDAO partitionedTableInfoDAO =
-        TableInfoDAO.builder()
-            .id(partitionedTableId)
-            .name(partitionedTableName)
-            .schemaId(UUID.fromString(schemaId))
-            .comment("Partitioned table")
-            .columns(List.of(firstName, age, country))
-            .dataSourceFormat(DataSourceFormat.DELTA.getValue())
-            .type(TableType.EXTERNAL.getValue())
-            .createdAt(new Date())
-            .updatedAt(new Date())
-            .url(partitionedTablePath)
-            .build();
-    partitionedTableInfoDAO
-        .getColumns()
-        .forEach(columnInfoDAO -> columnInfoDAO.setTable(partitionedTableInfoDAO));
-
-    try (Session session = factory.openSession()) {
-      Transaction tx = session.beginTransaction();
-      session.persist(partitionedTableInfoDAO);
-      tx.commit();
-    }
 
+    TableRepository tableRepository = TableRepository.getInstance();
+
+    CreateTable createExternalPartitionedTable = new CreateTable();
+    createExternalPartitionedTable.setName("user_countries");
+    createExternalPartitionedTable.setComment("Partitioned table");
+    createExternalPartitionedTable.setTableType(TableType.EXTERNAL);
+    createExternalPartitionedTable.setDataSourceFormat(DataSourceFormat.DELTA);
+    createExternalPartitionedTable.setStorageLocation(partitionedTablePath);
+    createExternalPartitionedTable.setCatalogName("unity");
+    createExternalPartitionedTable.setSchemaName("default");
+
+    // Add columns
+    ColumnInfo firstName = new ColumnInfo();
+    firstName.setName("first_name");
+    firstName.setTypeName(ColumnTypeName.STRING);
+    firstName.setTypeText(ColumnTypeName.STRING.getValue());
+    firstName.setTypeJson(ColumnTypeName.STRUCT.getValue());
+    firstName.setComment("string column");
+    firstName.setPosition(0);
+    createExternalPartitionedTable.addColumnsItem(firstName);
+
+    ColumnInfo age = new ColumnInfo();
+    age.setName("age");
+    age.setTypeName(ColumnTypeName.LONG);
+    age.setTypeText(ColumnTypeName.LONG.getValue());
+    age.setTypeJson(ColumnTypeName.LONG.getValue());
+    age.setComment("long column");
+    age.setPosition(1);
+    createExternalPartitionedTable.addColumnsItem(age);
+
+    ColumnInfo country = new ColumnInfo();
+    country.setName("country");
+    country.setTypeName(ColumnTypeName.STRING);
+    country.setTypeText(ColumnTypeName.STRING.getValue());
+    country.setTypeJson(ColumnTypeName.STRING.getValue());
+    country.setComment("partition column");
+    country.setPosition(2);
+    country.setPartitionIndex(0);
```

Comment:
why is partition index only being set for `country` column.

Reply:
As I understood [from the answer](https://github.com/unitycatalog/unitycatalog/discussions/217#discussioncomment-10031239), it is how partitioning information is handled by UC. Like it is the first partition column and if I would like to add the nested partitioning, the next column will have an index `1`.

Reply:
@tdas  can you please review this one ?